.github/workflows/check-apps.yml

```diff
@@ -50,37 +52,67 @@ jobs:
             ${{ runner.os }}-yarn
 
       - name: Get dependencies
+        if: ${{ steps.yarn-cache.outputs.cache-hit != 'true' }}
         run: yarn install --immutable
-  lint:
+
+  generate-client:
     if: github.event.pull_request.draft == false
     runs-on: ubuntu-22.04
     needs: deps
     defaults:
       run:
-        working-directory: apps
+        working-directory: ./apps
     steps:
       - uses: actions/checkout@v4
+        with:
+          submodules: "recursive"
+
+      - name: Cache election contract client
+        id: election-contract-cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            ./apps/voting/src/__generated__/election-contract
+          key: ${{ runner.os }}-election-contract-${{ hashFiles('./contracts/concordium-governance-committee-election/src/lib.rs') }}-${{ env.DUMMY }}
+          restore-keys: |
+            ${{ runner.os }}-election-contract
```

Comment:
I've always been used to generated code _not_ being included in the repository. It's how we've done for generated clients for grpc as well wherever we used that. Generating the code also ensures that components remain compatible with eachother.

Reply:
I don't insist on this, but the reason I think it would be nice is that it makes development simpler and makes for a more self-contained project, and also allows tracking of changes.

For the bridge project I committed the generated Rust clients, and we commit the generated proto files in the Rust SDK for example. But the reasons for the latter are different (we want to avoid people having protobuf installed).

Reply:
But feel free to not do this.

Reply:
Hmm.. well that's also a good point and to be fair I would have been annoyed by having to install a toolchain to generate a client.

I will add the client to the repository, but will keep the CI step for overriding the generated client to check compatibility.